### PR TITLE
[ramda] split tests to separate modules: 'join' - 'mean'

### PR DESCRIPTION
**Pull request by devrelm**
#37710 attempted to split the tests for `@types/ramda` into separate files per function, but the PR was too large to be approved on its own. This PR is part of a new piecemeal approach to migrating the tests.

I recommend stepping through the PR commit-by-commit to make it easier to check that no individual tests were dropped or changed substantively.

This PR is the next of many to move the functions over in a more piecemeal fashion. It continues from the following PRs:

* #38157 `[ramda] split tests to separate modules - functions beginning with 'a'`
* #38251 `[ramda] split tests to separate modules - functions beginning with 'b' - 'c'`
* #38274 `[ramda] split tests to separate modules - functions beginning with 'd' - 'e'`
* #38347 `[ramda] split tests to separate modules: 'filter' - 'head'`
* #38546 `[ramda] split tests to separate modules: 'identical' - 'isNaN'`

I'm splitting out the tests into separate files per function to make it easier to track what is and is not being tested. Please read the descriptions for #37710 and #38157 for more details.

---

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [ ] ~~Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>~~
- [ ] ~~If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.~~
- [ ] ~~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.~~
 P

### CODE REVIEW

**Review by drewwyatt — APPROVED** [review]
LGTM

File: types/ramda/ramda-tests.ts [hunk 1]
```diff
@@ -1077,10 +1070,6 @@ type Pair = KeyValuePair<string, number>;
     R.over(xyLens, R.negate, testObj);  // => {x: [{y: -2, z: 3}, {y: 4, z: 5}]}
 };
 
-() => {
-    R.keys({a: 1, b: 2, c: 3}); // => ['a', 'b', 'c']
```
[the remark this answers is not in the document]
Comment (devrelm) [reply types/ramda/ramda-tests.ts]: > I'm guessing it's intentional since it's literally the same test as the first one above.

exactly.

**Comment by devrelm** [comment]
@minestarks one more down, ~6 to go.

**Comment by devrelm** [comment]
Rebased to pick up changes to `keys` tests from #38467.

**Pull request merged**